src/rkopenmdao/file_writer.py

```diff
@@ -0,0 +1,97 @@
+"""Defines interface for file writers, as well as the HDF5 file writer."""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from itertools import chain
+
+import h5py
+
+# pylint doesn't pick it up, but it's there
+# pylint: disable = no-name-in-module
+from mpi4py.MPI import Comm
+import numpy as np
+
+from .metadata_extractor import TimeIntegrationMetadata
+
+
+@dataclass
+class FileWriterInterface(ABC):
+    """Interface for general file writers in RKOpenMDAO."""
+
+    file_name: str
+    time_integration_metadata: TimeIntegrationMetadata
+    comm: Comm
+
+    @abstractmethod
+    def setup_file(self):
```

Comment:
This violates the [RAII](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) principle. Clients always have to remember that they have to call this method ore their file writer is not going to work. I understand that it is quite easy to end up at this design as OpenMDAO is (sadly) full of it. I still recommend that we remove this method in the interface and let implementation classes do there setup in `__init__()` or [`__post__init__()`](https://docs.python.org/3/library/dataclasses.html#dataclasses.__post_init__).